### PR TITLE
Add script to build product and code sign

### PR DESCRIPTION
**Pull request by adashen**
The script did the following
1. Build modules under utils
2. Code sign the internal jars
3. install the signed jars
4. Build eclipse and inject the product eclipse appinsight key
5. Build intelliJ and inject the product intelliJ key
6. unzip the final packages and copy to the toSign folder

### CODE REVIEW

File: BuildProduct.sh [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        ?) echo "error: option -$OPTARG is not implemented"; exit ;;
+    esac
+done
+
+
+ECLIPSE_KEY=${ECLIPSE_TELKEY}
+INTELLIJ_KEY=${INTELLIJ_TELKEY}
+ARTIFACTS_DIR="artifacts"
+# check dir exists
+if [ ! -d  "$ARTIFACTS_DIR" ]; then
```
Comment (Eskibear) [inline BuildProduct.sh]: use absolute path

File: BuildProduct.sh [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+set -x
+
+# Build Utils
+mvn install -f $SCRIPTPATH/Utils/pom.xml -Dmaven.repo.local=$SCRIPTPATH/.repository $MAVEN_QUIET
+rm -rf /c/Signing/ToSign/*
+cp $SCRIPTPATH/Utils/azuretools-core/target/azuretools-core-${VERSION}.jar /c/Signing/ToSign/
+cp $SCRIPTPATH/Utils/azure-explorer-common/target/azure-explorer-common-${VERSION}.jar /c/Signing/ToSign/
+cp $SCRIPTPATH/Utils/hdinsight-node-common/target/hdinsight-node-common-${VERSION}.jar /c/Signing/ToSign/
+
+if /c/Signing/CodeSignUtility/csu.exe -c1=170
```
Comment (Eskibear) [inline BuildProduct.sh]: Path like `/c/Signing/...` depends on git bash for windows

File: BuildProduct.sh [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    mkdir -p $ARTIFACTS_DIR
+fi
+
+# echo shell commands when they are executed.
+set -x
+
+# Build Utils
+mvn install -f $SCRIPTPATH/Utils/pom.xml -Dmaven.repo.local=$SCRIPTPATH/.repository $MAVEN_QUIET
+rm -rf /c/Signing/ToSign/*
+cp $SCRIPTPATH/Utils/azuretools-core/target/azuretools-core-${VERSION}.jar /c/Signing/ToSign/
```
Comment (testforstephen) [inline BuildProduct.sh]: Use two variables to define ToSign and Signed directory. And reference these variables at later code.

File: BuildProduct.sh [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+# Build intellij 2017 plugin
+if [ $INTELLIJ_VERSION == "true" ] ; then
+    ./tools/IntellijVersionHelper 2017.1
+fi
+./gradlew clean buildPlugin --project-dir ./PluginsAndFeatures/azure-toolkit-for-intellij -s -Papplicationinsights.key=${INTELLIJ_KEY}
+cp ./PluginsAndFeatures/azure-toolkit-for-intellij/build/distributions/azure-toolkit-for-intellij.zip ./$ARTIFACTS_DIR/azure-toolkit-for-intellij-2017.zip
+
+# Extract jars to sign
+# intelliJ
+rm -rf /c/jenkins/toSignPackage/intelliJ/*
```
Comment (testforstephen) [inline BuildProduct.sh]: set /c/jenkins/toSignPackage to a variable toSignExtractPage

**Pull request merged**